pkg/controlplane/control/manager.go

```diff
@@ -138,6 +151,93 @@ func (m *Manager) DeleteImport(ctx context.Context, name types.NamespacedName) e
 		})
 }
 
+// addExport defines a new route target for ingress dataplane connections.
+func (m *Manager) addExport(ctx context.Context, export *v1alpha1.Export) (err error) {
+	m.logger.Infof("Adding export '%s/%s'.", export.Namespace, export.Name)
+
+	defer func() {
+		validCond := &metav1.Condition{
+			Type:   v1alpha1.ExportValid,
+			Status: metav1.ConditionTrue,
+			Reason: "Verified",
+		}
+
+		if err != nil {
+			validCond.Status = metav1.ConditionFalse
+			validCond.Reason = "Error"
+			validCond.Message = err.Error()
+		}
+
+		conditions := &export.Status.Conditions
+		if conditionChanged(conditions, validCond) {
+			meta.SetStatusCondition(conditions, *validCond)
+
+			m.logger.Infof(
+				"Updating export '%s/%s' status: %v.",
+				export.Namespace, export.Name, *conditions)
+			statusError := m.client.Status().Update(ctx, export)
+			if statusError != nil {
+				if err == nil {
+					err = statusError
+					return
+				}
+
+				m.logger.Warnf("Error updating export status: %v.", statusError)
+				return
+			}
+		}
+
+		if errors.Is(err, &exportServiceNotExistError{}) {
+			err = reconcile.TerminalError(err)
+		}
+	}()
+
+	if export.Spec.Host != "" {
+		return nil
+	}
+
+	name := types.NamespacedName{
+		Name:      export.Name,
+		Namespace: export.Namespace,
+	}
+
+	if err := m.client.Get(ctx, name, &v1.Service{}); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+
+		return exportServiceNotExistError{name}
+	}
+
+	return nil
+}
+
+// addService adds a new service.
+func (m *Manager) addService(ctx context.Context, service *v1.Service) error {
+	return m.checkExportService(ctx, types.NamespacedName{
+		Namespace: service.Namespace,
+		Name:      service.Name,
+	})
+}
+
+// deleteService deletes a service.
+func (m *Manager) deleteService(ctx context.Context, name types.NamespacedName) error {
+	return m.checkExportService(ctx, name)
```

Comment:
I Didn't understand why in delete you use checkExportService

Reply:
Because if a service was deleted, I want to check the relevant export (which will possibly make the export invalid)